doc/adr/0012-modules-structure.md

```diff
@@ -0,0 +1,94 @@
+
+The second (and main) reason for restructuring modules is the fact that 3rd parties cannot use the SDK module in the current form. It does not expose all required operations, so internal modules were used directly, leaking the project's internal structure and locking further internal changes.
+
+## Decision
+
+`keytool` module will be splitted into: operations/functions that we will put to `shared` and exposing an app that will be combined into `wallet` module.
+Then, all the things related to node operations would be kept in `node-shared` module (including gossip protocol, consensus, etc.). This leads to emptying the `sdk` module, so we will be able to implement the `sdk` module as a proxy between the external world and the internal tessellation structure. `sdk` module should reexpose all necessary operations to be used by 3rd parties. For example - to create a keypair, global L0 can use the `shared` module directly, but 3rd party will use the `sdk` module that will reexpose keypair creation function. That way, the internal structure of tessellation will be hidden behind the facade.
+
+Modules dependencies and libraries cannot be transient. We will use `% Optional` scope for all internal modules.
+
+- `shared` will contain base classes and extensions (like kryo or refined extensions), all the stuff used within the GlobalSnapshot (like Address, base Block or Transaction), base StateChannels types and key-related operations (like getting address from public key),
+- `wallet` will contain operations required by the wallet-cli (like creating transactions), but also will expose key operations (like creating a new key),
+- `dag-shared` will contain shared types and classes between Global L0 and DAG L1, related to $DAG (like $DAG block or $DAG transaction),
+- `node-shared` will contain all the operations required to launch a node (like base TessellationIO app, gossip protocol, cluster management or joining algorithm),
+- `global-l0` will be renamed from `core` and will remain the Global L0 module,
+- `dag-l1` will remain the DAG L1 module.
+- `sdk` 
+
+- `shared` - base internal library containing `GlobalSnapshot` schema as well as extensions for 3rd party libraries
+- `dag-shared` - an internal library containing $DAG-related classes (like $DAG block or $DAG transaction)
+- `node-shared` - an internal library containing all the classes necessary to launch a node (like base TessellationIO app, gossip protocol, cluster management or joining algorithm)
+- `wallet` - an app for creating transactions and managing keys
+- `global-l0` - the Global L0 node app (renamed from `core`)
+- `dag-l1` - the DAG L1 node app (remains unchanged)
+- `sdk` - a library exposing a set of functionalities as a Tessellation public interface (facade) for 3rd parties
+- `tools` - an internal app used for generating transactions, state channel snapshots and other things needed for development, maintenance or testing
```

Comment:
I like new definitions, but you kept both lists: the old one and the new one. I think we can keep only the new one.